packages/next-app/src/app/pool/[poolId]/(components)/TransactionDialog.tsx

```diff
@@ -0,0 +1,225 @@
+"use client";
+import * as Alert from "@radix-ui/react-alert-dialog";
+import cn from "classnames";
+import { useContext, useEffect, useState } from "react";
+import {
+  ImCheckboxChecked,
+  ImCheckboxUnchecked,
+  ImHourGlass,
+  ImWarning,
+} from "react-icons/im";
+import useSWR from "swr";
+
+import { Button } from "#/components";
+import {
+  PoolMetadataAttribute,
+  PoolMetadataContext,
+  UpdateStatus,
+} from "#/contexts/PoolMetadataContext";
+import { pinJSON } from "#/lib/ipfs";
+import metadataGql from "#/lib/poolMetadataGql";
+import { fetcher } from "#/utils/fetcher";
+import { writeSetPoolMetadata } from "#/wagmi/setPoolMetadata";
+
+enum ButtonLabels {
```

Comment:
not sure if I like this `enum` loosely defined here. If we change `updateStatus` then we might have to change something here as well. I would try to tie them to avoid mistakes.

Reply:
Refactored ✌🏻